vllm/model_executor/model_loader/loader.py

```diff
@@ -576,6 +575,169 @@ def save_model(
+            tensor_index = json.load(f)
+        
+        device_uuid_map = get_device_uuid_map()
+        device_uuid = device_uuid_map[rank]
+        replica_uuid = str(uuid.uuid4())
+        client = SllmStoreClient("localhost:8073")
+        
+        # sllm_state_dict = load_dict(os.path.join(local_model_path, f"rank_{rank}", device_config.device))
+        
+        with set_default_torch_dtype(model_config.dtype):
+            with torch.device(device_config.device):
+                model = _initialize_model(model_config, self.load_config,
+                                          lora_config, vision_language_config,
+                                          cache_config)
+            state_dict = self._filter_subtensors(model.state_dict())
+            
+            memory_ptrs = {rank: []}
+            tensor_copy_chunks = {rank: []}
+            for idx, (name, param) in enumerate(state_dict.items()):
+                data_ptr = param.untyped_storage().data_ptr()
+                memory_ptrs[rank].append(data_ptr)
+                offset, size, _, _, _ = tensor_index[name]
+                
+                # every tensor has its own base address, so GPU offset is always 0
+                tensor_copy_chunks[rank].append(
+                    (offset, size, 0, idx)
+                )
+            cuda_memory_handles = get_cuda_memory_handles(memory_ptrs)    
+            
+            ret = client.load_into_gpu(
+                model_name,
+                replica_uuid,
+                {device_uuid: tensor_copy_chunks[rank]},
+                {device_uuid: cuda_memory_handles[rank]}
+            )
+            if not ret or ret == False:
+                raise ValueError(f"Failed to load model {model_name} into GPU")
+            client.confirm_model_loaded(model_name, replica_uuid)
+        return model.eval()
+
+    @staticmethod
+    def save_model(
+        model: torch.nn.Module,
+        path: str,
+        pattern: Optional[str] = None,
+        max_size: Optional[int] = None,
+    ) -> None:
+        from vllm.distributed import get_tensor_model_parallel_rank
+        from serverless_llm_store._C import save_tensors
+        
+        # if pattern is None:
+        #     pattern = ShardedStateLoader.DEFAULT_PATTERN
+        rank = get_tensor_model_parallel_rank()
+        state_dict = ServerlessLLMLoader._filter_subtensors(model.state_dict())
+        
+        # move all tensors to CPU
+        for key, tensor in state_dict.items():
+            state_dict[key] = tensor.cpu()
+        
+        tensor_names = list(state_dict.keys())
+        tensor_data_index = {}
+        for name, param in state_dict.items():
+            param_storage = param.untyped_storage()
+            data_ptr = param_storage.data_ptr()
+            size = param_storage.size()
+            tensor_data_index[name] = (data_ptr, size)
+        
+        print(tensor_data_index)
+        rank_path = path + f"rank_{rank}" 
```

Comment:
local path seems different compared with load part. I guess it should be ` os.path.join(path, f"rank_{rank}")`?